core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -329,6 +334,20 @@ class TaskMetrics private[spark] (initialAccums: Seq[Accumulator[_]]) extends Se
 
   private[spark] def registerAccumulator(a: Accumulable[_, _]): Unit = {
     accums += a
+    if (a.consistent) {
+      consistentAccums += a
```

Comment:
oh this is a nice trick for avoiding the overhead when "consistency" isn't requested, I hadn't thought of this.